spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/couchbase/embedded/EmbeddedCouchbaseAutoConfiguration.java

```diff
@@ -0,0 +1,151 @@
+
+		CouchbaseProperties.Bucket bucketProperties = this.properties.getBucket();
+
+		BucketConfiguration bucketConfiguration = new BucketConfiguration();
+		bucketConfiguration.name = bucketProperties.getName();
+		bucketConfiguration.password = bucketProperties.getPassword();
+		bucketConfiguration.type = Bucket.BucketType.COUCHBASE;
+		bucketConfiguration.numReplicas = 1;
+		bucketConfiguration.numNodes = 1;
+
+		CouchbaseMock couchbaseMock = new CouchbaseMock(0, Collections.singletonList(bucketConfiguration));
+		couchbaseMock.start();
+		couchbaseMock.waitForStartup();
+
+		String host = couchbaseMock.getHttpHost();
+		int bootstrapHttpDirectPort = couchbaseMock.getHttpPort();
+		int bootstrapCarrierDirectPort = couchbaseMock.getBuckets().get(this.properties.getBucket().getName()).getServers()[0].getPort();
+
+		couchbaseMock.startHarakiriMonitor(host + ":" + bootstrapHttpDirectPort, false);
+
+		this.properties.setBootstrapHosts(Collections.singletonList(host));
+		this.properties.setBootstrapHttpDirectPort(bootstrapHttpDirectPort);
+		this.properties.setBootstrapCarrierDirectPort(bootstrapCarrierDirectPort);
+
+		setEnvironmentProperties(this.applicationContext, host, bootstrapHttpDirectPort, bootstrapCarrierDirectPort);
+
+		return couchbaseMock;
+	}
+
+	private void setEnvironmentProperties(ApplicationContext currentContext,
+			String host,
+			int bootstrapHttpDirectPort,
+			int bootstrapCarrierDirectPort) {
+		if (currentContext instanceof ConfigurableApplicationContext) {
+			MutablePropertySources sources = ((ConfigurableApplicationContext) currentContext)
+					.getEnvironment().getPropertySources();
+			Map<String, Object> couchbaseProperties = getCouchbase(sources);
+			couchbaseProperties.put("spring.couchbase.bootstrap-hosts", host);
+			couchbaseProperties.put("spring.couchbase.bootstrap-http-direct-port", bootstrapHttpDirectPort);
+			couchbaseProperties.put("spring.couchbase.bootstrap-carrier-direct-port", bootstrapCarrierDirectPort);
+		}
+		if (currentContext.getParent() != null) {
+			setEnvironmentProperties(currentContext.getParent(), host, bootstrapHttpDirectPort, bootstrapCarrierDirectPort);
+		}
+	}
+
+	private Map<String, Object> getCouchbase(MutablePropertySources sources) {
+		PropertySource<?> propertySource = sources.get("spring.couchbase");
+		if (propertySource == null) {
+			propertySource = new MapPropertySource("spring.couchbase", new HashMap<>());
+			sources.addFirst(propertySource);
+		}
+		return (Map<String, Object>) propertySource.getSource();
+	}
+
+
+	/**
+	 * Additional configuration to ensure that {@link CouchbaseEnvironment} beans depend on the
+	 * {@code couchbaseMock} bean.
+	 */
+	@Configuration
+	@ConditionalOnClass({CouchbaseEnvironment.class})
+	protected static class EmbeddedCouchbaseDependencyConfiguration
+			implements BeanFactoryPostProcessor {
+
+		@Override
+		public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
```

Comment:
That's quite fragile IMO. There is no guarantee that `CouchbaseProperties` won't be processed differently in the future.

Reply:
removed this part